lib/mail_catcher.rb

```diff
@@ -197,74 +178,89 @@ def run! options=nil
     end
 
     puts "Starting MailCatcher"
+    puts "==> #{smtp_url}"
+    puts "==> #{http_url}"
 
-    Thin::Logging.debug = development?
-    Thin::Logging.silent = !development?
+    Async.logger.level = Logger::DEBUG if options[:verbose]
 
-    # One EventMachine loop...
-    EventMachine.run do
-      # Set up an SMTP server to run within EventMachine
-      rescue_port options[:smtp_port] do
-        EventMachine.start_server options[:smtp_ip], options[:smtp_port], Smtp
-        puts "==> #{smtp_url}"
+    if options[:daemon]
+      if quittable?
+        puts "*** MailCatcher runs as a daemon by default. Go to the web interface to quit."
+      else
+        puts "*** MailCatcher is now running as a daemon that cannot be quit."
       end
+      Process.daemon
```

Comment:
Daemonized the server here, as daemonizing in the `async.reactor` block shuts down the process. so doing it prior works well.

Reply:
If the daemonization is moved up here then the ports need to be opened first so that port collisions can be reported on the terminal.

Reply:
Indeed, Ill open the port first and report it on the terminal.